### DISCUSSION TITLE
how would this compare with js native libraries like ChartJs or Apex Charts

### DISCUSSION THREAD

**Discussion by mingsterism** [mhkeller/layercake#225] [Q&A] [answered]
hi, 
Would like to understand the implications of using this library vs other more established libraries like ChartJS. 

Is there any benefits to using this svelte native library. 

Thanks

**Comment by mhkeller** [comment, marked as answer]
The main difference is in the interface and what Layer Cake allows the user to do. From the [guide](https://layercake.graphics/guide):

> Layer Cake is more about having a system to organize your own custom components than it is a high-level charting library.

Looking at those two libraries you mentioned, it seems like you provide data and a JSON specification and you get a full chart with you line/bar/etc. as well as axes and tooltips. This is different from Layer Cake's design goals. The goals of Layer Cake are to give you a common set of scales and SVG, HTML and canvas components and let you populate them with renderers of your choosing. This lets users create fully customizable graphics.

The website includes examples of different kinds of renderers – viewable in the [Component Gallery](https://layercake.graphics/components) – but it does not come with any renderers built-in. 

If you want to make custom graphics with fully control over design and interactivity, then Layer Cake is aligned with that.
